### ISSUE TITLE
Enabling biometric lock prevents app from functioning

### ISSUE THREAD

**Issue by ZenerDeveloper** [Akylas/com.akylas.documentscanner#56]
**Device and App information:**
S21 Ultra 5g
Android 13 June 1st security patch
App version 1.2.12 Build 22 (Originally build 21 from Izzyondroid, Updated to 22 from github)

**Issue**
After enabling both biometrics and automatic biometric lock, the app won't open at all. It briefly shows the login prompt but quickly closes. Logs don't reveal much.

If I tap on the icon and the cancel button location at the same time, it will occasionally open but no documents are shown and no buttons work.
<details><summary>Screen Captures</summary>
<p>

https://github.com/Akylas/com.akylas.documentscanner/assets/80724816/8c46a33f-b9e4-4282-b4da-668a861c1923

If I tap on the icon and the cancel button location at the same time, it will occasionally open but no documents are shown and no buttons work.

![Screenshot_20231228_231329_OSS Document Scanner](https://github.com/Akylas/com.akylas.documentscanner/assets/80724816/783267fb-381f-4a7d-a017-d4448af3e1f8)

</p>
</details> 

I can provide any logs if you think they would prove useful, but again not much is shown (app starts, biometrics start, app closes, biometrics cancelled)

**Comment by farfromrefug** [comment]
@ZenerDeveloper sorry for the trouble will fix that asap. Can you get logcat, can you share it?

**Comment by ZenerDeveloper** [comment]
I installed that version of the app and I believe it should have sent over logs. I'm wondering if it has to do with the way Samsung implements their fingerprint reader since it is under screen and has to show a dialog over the app. However, I created a simple biometric app from code on the internet and it opens and recognizes just fine.

**Comment by farfromrefug** [comment]
@ZenerDeveloper sadly I dont see the error report. not sure why. can you get a logcat ? (using adb)

**Comment by ZenerDeveloper** [comment]
I have attached a log gathered with Logfox (an android app). It should contain everything immediately before and after the app was opened and subsequently closed. 
[29_12-14-06-51_779.log](https://github.com/Akylas/com.akylas.documentscanner/files/13795662/29_12-14-06-51_779.log)
I'm curious if it is an issue with your app specifically or if it is an issue with the biometric plugin you're using. I couldn't find any other apps using the plugin so I have no way to test that theory.

**Comment by farfromrefug** [comment]
@ZenerDeveloper Which version did you try that with? It is weird cause i dont see any log related to the app (i mean logs from the app). Did you use  https://github.com/Akylas/com.akylas.documentscanner/releases/tag/test_android  ?

**Comment by ZenerDeveloper** [comment]
Sorry, I had reinstalled the other build while messing around with it. Here is another log but with the test build installed.
[29_12-15-19-28_581.log](https://github.com/Akylas/com.akylas.documentscanner/files/13795877/29_12-15-19-28_581.log)

**Comment by farfromrefug** [comment]
@ZenerDeveloper it seems like the fingerprint recognition is "cancelled". This is the event i receive. And if this happens i need to close the app (for security reason).
Not sure why though. 
i will try with more phone once i get back home. I have another samsung device. Hopefully i will be able to reproduce the bug. In the meantime i am sorry but i dont know what to do

**Comment by farfromrefug** [comment]
@ZenerDeveloper I released a new version which hopefully will fix your issue

**Comment by ZenerDeveloper** [comment]
The issue is "fixed" as in the app will not self close. However, all thumbnails are visible without unlocking the app, mostly defeating the purpose of the feature.

**Comment by farfromrefug** [comment]
@ZenerDeveloper I was not sure what to do about this. was not sure blurring or anything was needed. but if you say that then it is necessary ;)

**Comment by ZenerDeveloper** [comment]
I figured users may be scanning sensitive documents (tax returns perhaps) that they'd want behind a fingerprint lock and having the thumbnail shown when locked could defeat that if say a social security number, credit card number, phone number etc. Were to be present. If a device was left unlocked say on a desk, someone could theoretically view some sensitive information.

Tl;dr, it would be appreciated to leave the homescreen blank / blurred until unlocked to prevent simple attacks against personal info.

**Comment by farfromrefug** [comment]
@ZenerDeveloper i  just implemented to a full screen view ensuring you dont see the data while requesting biometrics.
Now i need to add an option to prevent screenshots

**Comment by farfromrefug** [comment]
@ZenerDeveloper should all be implemented in latest version

**Comment by ZenerDeveloper** [comment]
Yes it all works perfectly now. Thank you I will mark the issue closed.
